src/queue.jl

```diff
@@ -6,31 +6,57 @@ Each queue is uniquely associated with an agent.
 """
 mutable struct HSAQueue
     agent::HSAAgent
-    queue::Ref{Ptr{HSA.Queue}}
+    queue::Ptr{HSA.Queue}
     active::Bool
 end
-
+const QUEUES = WeakKeyDict{Ptr{HSA.Queue},HSAQueue}()
 const DEFAULT_QUEUES = IdDict{HSAAgent,HSAQueue}()
 
+function queue_error_handler(status::HSA.Status, _queue::Ptr{HSA.Queue}, data::Ptr{Cvoid})::Nothing
+    # TODO: From which thread are we calling this?
+    if status !== HSA.STATUS_SUCCESS then
+        queue = QUEUES[_queue]
+        queue.active = false
+        @assert DEFAULT_QUEUES[queue.agent] == queue # @jpsamaroo who gurantuees this 1:1 relation?
```

Comment:
This is wrong, the default queue isn't necessarily the one calling this handler. This is managed by the `HSAAgent` ctor, IIRC, as well as the launch code, which will create a new default queue if one doesn't exist.